geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/Message.java

```diff
@@ -1161,10 +1288,146 @@ public void receive() throws IOException {
   public void receive(ServerConnection sc, int maxMessageLength, Semaphore dataLimiter,
       Semaphore msgLimiter) throws IOException {
     this.serverConnection = sc;
+    if (sc.getIOFilter() != null) {
```

Comment:
I see this pattern in many places. It seems like the null check could be moved into setIOFilter which would simplify all the callers.